### ISSUE TITLE
linter: no-unused-vars reports commented out JSX variables as used when referenced in return statement of a React component

### ISSUE THREAD

**Issue by taearls** [oxc-project/oxc#8225]
### What version of Oxlint are you using?

latest

### What command did you run?

npx oxlint

### What does your `.oxlint.json` config file look like?

No config (no_unused_vars warns in the default config)

### What happened?

I was playing around with the linter, and I noticed that if a variable is referenced in a comment inside the return statement of a JSX component, it is not considered unused.

Snippet:

```tsx
import React from 'react'

const Foo = () => '';
const Bar = () => '';

const DummyComponent: React.FC = () => {

  // Bar is not being used here, but it is correctly reported as unused.
  // <Bar />
  return (
    <div>
    // Foo is not being used here, but it is incorrectly reported as used.
    // <Foo />
    </div>
  )
}

export default DummyComponent
```

Playground with minimal reproduction:
https://playground.oxc.rs/#eNqFkcFqwzAMhl9F+JIWQrMNdsnaHtbRB9i5FydROoNtFdnuGkrefXZSFzYGhUB+Kb//T3KuohW1UOZE7OETZeuhZzJQcNLFwR5sS9Z52BPBBhZL2GyhKN5y+13y73b+8BGMGXYUgy1aX8/Zq/3u7r4mL0BVTRnKgSUPDSp7hOCwgy9kLKEJHlR8HLTEjK3XAzCmaaNFOgg2mVe3pHWKqrapYvSBLSySBlh36jy1J1va5TFQxU3+RWbgjExhMzJiqsxZHuyYFsTLdLMd9jLov7ciSkGivgoONr3cYL28iNpzwFJoFQ037Vo64b0YTEM6V56ldT2xEXUvtcOxFCfJDjkmRp1Ssr5bp4bwko8YEQLdy9Pzq4iOljo8YhomFkZZ1at8OP5Vz6T3mr7T4TNyQy7ONEPH8Qdni8Lx

**Comment by camc314** [comment]
this is working as expected. // doesn't comment out code in JSX. you have to use an a JSXExpression and comment inside that.

**Comment by taearls** [comment]
ah, I see. you're right, my mistake. thanks!
